-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-35327/25370

ERROR: Build errors found during clang-tidy run.
```
  theTTopoToken = esConsumes();
  ^
RecoMuon/TrackerSeedGenerator/plugins/TSGFromL2Muon.cc:46:19: error: no matching member function for call to 'esConsumes' [clang-diagnostic-error]
  theTTopoToken = esConsumes();
                  ^
/cvmfs/cms-ib.cern.ch/nweek-02698/slc7_amd64_gcc900/cms/cmssw-patch/CMSSW_12_1_X_2021-09-17-1100/src/FWCore/Framework/interface/EDConsumerBase.h:223:34: note: candidate function template not viable: no known conversion from 'TSGFromL2Muon' to 'edm::EDConsumerBase' for object argument
--
RecoMuon/TrackerSeedGenerator/plugins/TSGFromL2Muon.cc:64:46: error: use of undeclared identifier 'theTTopoToken' [clang-diagnostic-error]
  const TrackerTopology* tTopo = &es.getData(theTTopoToken);
                                             ^
RecoMuon/TrackerSeedGenerator/plugins/TSGFromL2Muon.h:41:8: error: C++ requires a type specifier for all declarations [clang-diagnostic-error]
  edm::ESTrackerTopologypology, TrackerTopologyRcd > theTTopoToken;
       ^
RecoMuon/TrackerSeedGenerator/plugins/TSGFromL2Muon.h:41:8: error: non-friend class member 'ESTrackerTopologypology' cannot have a qualified name [clang-diagnostic-error]
RecoMuon/TrackerSeedGenerator/plugins/TSGFromL2Muon.h:41:33: error: C++ requires a type specifier for all declarations [clang-diagnostic-error]
  edm::ESTrackerTopologypology, TrackerTopologyRcd > theTTopoToken;
                                ^
RecoMuon/TrackerSeedGenerator/plugins/TSGFromL2Muon.h:41:51: error: expected ';' at end of declaration list [clang-diagnostic-error]
  edm::ESTrackerTopologypology, TrackerTopologyRcd > theTTopoToken;
                                                  ^
                                                  ;
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:129: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/RecoMuon/TrackerSeedGenerator/plugins/TSGFromL2Muon.cc
+++ b/RecoMuon/TrackerSeedGenerator/plugins/TSGFromL2Muon.cc
@@ -42,6 +42,8 @@ TSGFromL2Muon::TSGFromL2Muon(const edm::ParameterSet& cfg) {
   //L2 collection
   theL2CollectionLabel = cfg.getParameter<edm::InputTag>("MuonCollectionLabel");
   l2muonToken = consumes<reco::TrackCollection>(theL2CollectionLabel);
+
+  theTTopoToken = esConsumes();
 }
 
 TSGFromL2Muon::~TSGFromL2Muon() = default;
@@ -59,9 +61,7 @@ void TSGFromL2Muon::produce(edm::Event& ev, const edm::EventSetup& es) {
   auto result = std::make_unique<L3MuonTrajectorySeedCollection>();
 
   //Retrieve tracker topology from geometry
-  edm::ESHandle<TrackerTopology> tTopoHand;
-  es.get<TrackerTopologyRcd>().get(tTopoHand);
-  const TrackerTopology* tTopo = tTopoHand.product();
+  const TrackerTopology* tTopo = &es.getData(theTTopoToken);
 
   //intialize tools
   theService->update(es);

--- a/RecoMuon/TrackerSeedGenerator/plugins/TSGFromL2Muon.h
+++ b/RecoMuon/TrackerSeedGenerator/plugins/TSGFromL2Muon.h
@@ -17,6 +17,7 @@ class MuonServiceProxy;
 class TrackerSeedGenerator;
 class MuonTrackingRegionBuilder;
 class TrackerSeedCleaner;
+class TrackerTopologyRcd;
 
 //
 // Generate tracker seeds from L2 muons
@@ -37,5 +38,6 @@ private:
   std::unique_ptr<TrackerSeedGenerator> theTkSeedGenerator;
   std::unique_ptr<TrackerSeedCleaner> theSeedCleaner;
   edm::EDGetTokenT<reco::TrackCollection> l2muonToken;
+  edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> theTTopoToken;
 };
 #endif

